### PR TITLE
Metrics WebApp 

### PR DESCRIPTION
**Pull request by peterdtitan**
Metrics App is an app designed to give each user an idea of what the air quality is like around supported cities of the world. It features a countries page with dynamic navigation to states and then cities within the country, before returning info on the selected city. It makes use of the redux store to update the countries as well as the state and cities for each country. Data on the city is fetched on each render as the values are constantly changed.

The app is live on [Metrics WebApp](https://metrics-app.onrender.com)
 
A live presentation of the app can also be viewed on: [Live Presentation](https://www.loom.com/share/ded06fb3da914ccabec519b06ec16bf8)

Below are some of the features of the app in use: 

Screenshot 2023-04-27 at 8 43 04 PM

Screenshot 2023-04-27 at 8 44 47 PM

Screenshot 2023-04-27 at 8 45 51 PM

### CODE REVIEW

File: src/components/__tests__/countriesIntegration.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { Provider, useSelector } from 'react-redux';
+import { MemoryRouter } from 'react-router-dom';
+import Countries from '../Countries';
+import store from '../../redux/store';
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+}));
+
```
Comment (DammyShittu) [inline src/components/__tests__/countriesIntegration.test.js:12]: - [x] Good job adding integration tests for your app with the react testing library, however, let's also add enough unit tests for the individual components like Cities, Countries, Info and States.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @peterdtitan,

## Highlights
✔️ Search feature

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/__tests__/countriesIntegration.test.js (hunk 1 above)
Comment (peterdtitan) [reply src/components/__tests__/countriesIntegration.test.js:12]: Okay. Thanks for the feedback!

**Comment by peterdtitan** [comment]
Added more tests as suggested by review. The following suites were created for: 

- [ ] Cities
- [ ] States
- [ ] Individual city info page

Screenshot 2023-04-28 at 11 12 11 AM

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
-  Great job implementing more test requested by the previous code reviewer. :heavy_check_mark: 
**OPTIONAL**
Please make sure you use the right [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) as required per project requirement.
This was missed by the previous code reviewer.

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**